trio/_socket.py

```diff
@@ -241,9 +292,14 @@ def from_stdlib_socket(sock):
 
 
 @_wraps(_stdlib_socket.fromfd, assigned=(), updated=())
-def fromfd(fd, family, type, proto=0):
+def fromfd(
+    fd: SupportsIndex,
+    family: AddressFamily | int = _stdlib_socket.AF_INET,
+    type: SocketKind | int = _stdlib_socket.SOCK_STREAM,
+    proto: int = 0,
+) -> _SocketType:
     """Like :func:`socket.fromfd`, but returns a Trio socket object."""
-    family, type, proto = _sniff_sockopts_for_fileno(family, type, proto, fd)
+    family, type, proto = _sniff_sockopts_for_fileno(family, type, proto, int(fd))
```

Comment:
I don't think `int(fd)` is guaranteed to be the same behavior as `fd.__index__()` but it's probably ... fine?

Reply:
We'd want `operator.index()` here. `int()` would allow floats/strings which probably isn't a good thing...